### ISSUE TITLE
AWSS3TransferUtility support for object metadata during downloads.

### ISSUE THREAD

**Issue by chandusam2002** [aws-amplify/aws-sdk-ios#2373]
**Which AWS Services is the feature request for?**
AWSS3 

**Is your feature request related to a problem? Please describe.**
AWSS3TransferUtility does not let you retrieve object metadata during a download. During the upload process AWSS3TransferUtilityUploadExpression allows uploading the object metadata. But the same support is not provided for downloads.

**Describe the solution you'd like**
The completionHandler for AWSS3TransferUtility could include a metadata block that can be used to retrieve the metadata. I don't know too much about the design of the library. There could be a better implementation.

**Describe alternatives you've considered**
I have been using AWSS3.s3 as an alternative. But TransferUtility supports backgrounding/resuming   options which  are useful for us.

**Additional context**
https://github.com/awsdocs/aws-mobile-developer-guide/blob/master/doc_source/how-to-transfer-files-with-transfer-utility.rst

**Comment by bumbleparrot** [comment]
Checking in on this.

Is the only way to get the metadata from an S3 item still to use the AWSS3 object's AWSS3.headObject(request) method?

So when I download an item from S3 (say an image), I have to make a SECOND network call just to get the associated metadata?

**Comment by palpatim** [comment]
That is correct. To my knowledge, S3 doesn't return metadata with object data, so the feature request is to either add a new S3 service API to return object data plus metadata, to make a convenience call to the `AWSS3.headObject` before fulfilling the TransferUtility download, or to expose a convenience `getMetadata` or similar function.
